### PR TITLE
Refactor typescript model generation to make local testing easier

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?:
Refactor typescript model generation to make local testing easier

### Which issue(s) this PR fixes:
No issue is created.

### PR acceptance criteria:
Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

> - Open new test/doc issues under the [devfile/api](https://github.com/devfile/api/issues) repo
> - Check each criteria if:
>  - There is a separate tracking issue. Add the issue link under the criteria
>  **or**
>  - test/doc updates are made as part of this PR
> -  If unchecked, explain why it's not needed

- [ ] Unit/Functional tests

  

- [ ] [QE Integration test](https://github.com/devfile/integration-tests) 

  

- [ ] Documentation 

   

- [ ] Client Impact

### CODE REVIEW

File: build/typescript-model/generate.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+SCRIPT_DIR=`dirname $( readlink -m $( type -p ${0} ))`
+WORK_DIR=${SCRIPT_DIR}/workdir
+echo "[INFO] Using the following folder to store all build files ${SCRIPT_DIR}/workdir"
+mkdir -p $WORK_DIR
+
+GEN_REVISION=5c6d90b260fd94af32157f304f971778c899b5e2
+
+k8s_client_gen() {
+    [ ! -d $WORK_DIR/gen ] && git clone https://github.com/kubernetes-client/gen.git $WORK_DIR/gen || echo "kubernetes-client/gen is already clonned into $WORK_DIR/gen"
```
Comment (JPinkney) [inline build/typescript-model/generate.sh:12]:
```suggestion
    [ ! -d $WORK_DIR/gen ] && git clone https://github.com/kubernetes-client/gen.git $WORK_DIR/gen || echo "kubernetes-client/gen is already cloned into $WORK_DIR/gen"
```

File: build/typescript-model/generate.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    yarn && yarn build || "[ERROR] Generated typescript model failed to build. Check it at $WORK_DIR/typescript-models."
+    echo "[INFO] Done."
+}
+
+generate_swagger_json
+k8s_client_gen
+build_typescript_model
+
+echo "[INFO] Typescript model is successfully generated and verified."
+echo "[INFO] It can be accessed at $WORK_DIR/typescript-models"
```
Comment (JPinkney) [inline build/typescript-model/generate.sh:64]: newline

File: .gitignore [hunk 3]
```diff
@@ -83,3 +83,5 @@ tags
 generator/build/generator
 #Mac file
 .DS_Store
+
+build/typescript-model/workdir
```
Comment (JPinkney) [inline .gitignore:87]: newline

File: build/typescript-model/generate.sh (hunk 1 above)
Comment (sleshchenko) [reply build/typescript-model/generate.sh:12]: Thx. Applied

File: .gitignore (hunk 3 above)
Comment (sleshchenko) [reply .gitignore:87]: thx. applied

File: build/typescript-model/generate.sh (hunk 2 above)
Comment (sleshchenko) [reply build/typescript-model/generate.sh:64]: thx. applied

**Review by JPinkney — APPROVED** [review]

**Pull request merged**